### PR TITLE
Quick Sort

### PR DESCRIPTION
**Pull request by ghost**
Added quick sort algorithm (package name fixed).

### CODE REVIEW

File: quicksort.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+public class MyQuickSort {
```
Comment (cegerard) [inline quicksort.java:1]: This class must implements the ISortingAlgorithm interface

src/main/java/sortBench/algorithms/ISortingAlgorithm.java

File: quicksort.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            quickSort(i, higherIndex);
+    }
+ 
+    private void exchangeNumbers(int i, int j) {
+        int temp = array[i];
+        array[i] = array[j];
+        array[j] = temp;
+    }
+     
+    public static void main(String a[]){
```
Comment (cegerard) [inline quicksort.java:56]: You can get rid of this static main method.
You will never use it in this project.

**Review by cegerard — CHANGES_REQUESTED** [review]

File: quicksort.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+public class MyQuickSort {
+     
+    private int array[];
```
Comment (cegerard) [inline quicksort.java:3]: private int[] array;

**Review by cegerard — COMMENTED** [review]
add the algorithm package like in the default algorithm

`package sortBench.algorithm;`

**Review by cegerard — COMMENTED** [review]
Put your algorithm in the right directory, like it was done with default sorting algorithm.

src/main/java/sortBench/algorithms/quickSort

**Comment by cegerard** [comment]
Remember, you have write access on this repository so you can clone it directly and work with pull/push 
It is not needed to fork it.

**Comment by ghost** [comment]
Got it all fixed !
(java path + package name + implementing + useless stuff)
(Just as well as mergesort.)

Should I open another request or is the pull request keeping itself up-to-date along with my new fork ?
(Sorry. Still having a better feeling with going through my GitHub fork somehow.)

**Comment by cegerard** [comment]
@UserPOE-Frk Yes your pull request stay in sync.

Test are broken because of your class naming
To keep it simple:

1. Remove the quickSort.java file from the root of this projet (the first one you create)
2. Put your quickSort.java into src/main/java/sortBench/algorithms/quickSort path (create a new directory named quickSort)
3. In Java the class must have the same name than its file and its first letter Capitalized (CamelCase for naming). Just rename your file and your class with `QuickSort`


After that it will be good

**Comment by ghost** [comment]
Noticed during post-tests a few hours then indeed. I've made it through that little time paradox !
Converted it all into CamelCase as well. Darn Windows messing up with it all the way...

**Comment by cegerard** [comment]
Ok I will validate this.
You forgot to delete the quicksort.java from root directory, but you can do it in another PR if you want.

**Review by cegerard — APPROVED** [review]

**Comment by cegerard** [comment]
So it is OK you can merge your modifications by clicking the green button 😄

**Pull request merged**